.github/workflows/tests.yaml

```diff
@@ -0,0 +1,37 @@
+name: "Tests"
+
+on:
+  pull_request:
+  push:
+    branches:
+      - 'master'
+    tags:
+      - '*'
+
+permissions:
+  pull-requests: write
+  contents: read
+  id-token: write
+
+env:
+  GOLANG_VERSION: "1.16"
```

Comment:
Should we use a newer version of Go here? Or run a matrix of tests with a few versions of Go?